iceoryx_utils/include/iceoryx_utils/cxx/helplets.hpp

```diff
@@ -292,6 +292,23 @@ using BestFittingType_t = typename BestFittingType<Value>::Type_t;
 #endif
 #endif
 
+/// @brief IOX_MAYBE_UNUSED adds the [[gnu::unused]] or [[maybe_unused]] attribute when it is available for the current
+/// compiler.
+/// @note
+///   activate attribute for gcc or clang
+#if defined(__GNUC__) || defined(__clang__)
+#define IOX_MAYBE_UNUSED [[gnu::unused]] // NOLINT
```

Comment:
Thanks for adding this! What do you think about removing the maybe and just calling it `IOX_UNUSED`? IMHO we should be explicit as possible, maybe can be misunderstood. It's not called `[[gnu::maybe_unused]]` either :P

Reply:
But the C++17 attribute is called [[maybe_unused]]